File: naïve.kt
fun nthPrime(n: Int): Int {
	var i = 0
    var j = 2
	while (i < n) {
		if (j.isPrime()) {
			i += 1
		}
		j += 1
	}
	return j - 1
}

fun Int.isPrime(): Boolean {
	for (i in 2 until this) {
        if (this % i == 0) {
            return false
        }
    }
	return true
}

fun main(args: Array<String>) {
    print(nthPrime(10000))
}

File: naïve.js
function nthPrime(n, j = 2) {
	for (let i = 0; i < n; j += 1)
		if (isPrime(j)) i += 1
	return j - 1
}

function isPrime(x) {
	if (x < 2) return false
	for (let i = x - 1; i >= 2; i -= 1) 
		if (x % i === 0) return false
	return true
}

console.log(nthPrime(10 ** 4))
